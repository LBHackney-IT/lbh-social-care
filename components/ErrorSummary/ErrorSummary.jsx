import PropTypes from 'prop-types';
import cx from 'classnames';

const ErrorSummary = ({ title, body, links, isRestricted }) => (
  <div
    className={cx('govuk-error-summary', {
      'restricted-summary-box': isRestricted,
    })}
    aria-labelledby="error-summary-title"
    role="alert"
    tabIndex="-1"
    data-module="govuk-error-summary"
  >
    <h2
      className={cx('govuk-error-summary__title', {
        'restricted-summary-title': isRestricted,
      })}
      id="error-summary-title"
    >
      {title}
    </h2>
    <div className="govuk-error-summary__body">
      {body}
      {links && links.length > 0 && (
        <ul className="govuk-list govuk-error-summary__list">
          {links.map(({ href, text }) => (
            <li key={href}>
              <a href={href}>{text}</a>
            </li>
          ))}
        </ul>
      )}
    </div>
  </div>
);

ErrorSummary.propTypes = {
  title: PropTypes.string,
  body: PropTypes.node,
};

export default ErrorSummary;
