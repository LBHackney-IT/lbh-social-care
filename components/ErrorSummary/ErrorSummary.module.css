.restricted_summary_box {
  border-top: 1px;
  border-bottom: 1px;
  border-right: 1px;
  background-color: #f8ecea;
}
