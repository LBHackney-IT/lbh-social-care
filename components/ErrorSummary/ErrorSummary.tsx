export interface Props {
  title?: string;
  body?: React.ReactChild;
  role?: string;
  links?: Array<{
    text: string;
    href: string;
  }>;
}

const ErrorSummary = ({
  title,
  body,
  links,
  role = 'alert',
}: Props): React.ReactElement => (
  <div
    className="govuk-error-summary"
    aria-labelledby="error-summary-title"
    role={role}
    tabIndex={-1}
    data-module="govuk-error-summary"
  >
    <h2 className="govuk-error-summary__title" id="error-summary-title">
      {title}
    </h2>
    <div className="govuk-error-summary__body">
      {body}
      {links && links.length > 0 && (
        <ul className="govuk-list govuk-error-summary__list">
          {links.map(({ href, text }) => (
            <li key={href}>
              <a href={href}>{text}</a>
            </li>
          ))}
        </ul>
      )}
    </div>
  </div>
);

export default ErrorSummary;
