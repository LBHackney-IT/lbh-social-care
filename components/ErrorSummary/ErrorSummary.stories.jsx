import ErrorSummary from './ErrorSummary';

export default {
  title: 'ErrorSummary',
  component: ErrorSummary,
};

const Template = (args) => <ErrorSummary {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Error Summary',
  body: 'i am an error summary box',
  links: [
    { href: 'foo', text: 'foo' },
    { href: 'bar', text: 'bar' },
  ],
};
