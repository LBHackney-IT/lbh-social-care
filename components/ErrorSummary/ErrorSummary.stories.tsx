import { Story } from '@storybook/react';

import ErrorSummary, { Props } from './ErrorSummary';

export default {
  title: 'ErrorSummary',
  component: ErrorSummary,
};

const Template: Story<Props> = (args) => <ErrorSummary {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Error Summary',
  body: 'i am an error summary box',
  links: [
    { href: 'foo', text: 'foo' },
    { href: 'bar', text: 'bar' },
  ],
};
