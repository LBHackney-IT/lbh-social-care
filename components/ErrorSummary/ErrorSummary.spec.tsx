import { render } from '@testing-library/react';
import ErrorSummary from './ErrorSummary';

describe('ErrorSummary component', () => {
  const props = {
    title: 'Error Summary',
    body: 'i am an error summary box',
    links: [
      { href: 'foo', text: 'foo' },
      { href: 'bar', text: 'bar' },
    ],
  };

  it('should render properly', () => {
    const { asFragment } = render(<ErrorSummary {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass through the role attribute', () => {
    const { getByRole } = render(
      <ErrorSummary {...props} role="complementary" />
    );
    getByRole('complementary');
  });
});
