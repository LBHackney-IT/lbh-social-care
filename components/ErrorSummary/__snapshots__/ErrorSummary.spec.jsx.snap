// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSummary component should render properly 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="error-summary-title"
    class="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabindex="-1"
  >
    <h2
      class="govuk-error-summary__title"
      id="error-summary-title"
    >
      Error Summary
    </h2>
    <div
      class="govuk-error-summary__body"
    >
      i am an error summary box
      <ul
        class="govuk-list govuk-error-summary__list"
      >
        <li>
          <a
            href="foo"
          >
            foo
          </a>
        </li>
        <li>
          <a
            href="bar"
          >
            bar
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
